solr/core/src/java/org/apache/solr/search/FunctionRangeQuery.java

```diff
@@ -25,18 +25,16 @@
 import org.apache.lucene.queries.function.ValueSource;
 import org.apache.lucene.queries.function.ValueSourceScorer;
 import org.apache.lucene.search.IndexSearcher;
-import org.apache.lucene.search.Query;
 import org.apache.lucene.search.QueryVisitor;
 import org.apache.lucene.search.ScoreMode;
 import org.apache.lucene.search.Weight;
 import org.apache.solr.common.SolrException;
 import org.apache.solr.search.function.ValueSourceRangeFilter;
 
 // This class works as either a normal constant score query, or as a PostFilter using a collector
-public class FunctionRangeQuery extends Query implements PostFilter {
+public class FunctionRangeQuery extends ExtendedQueryBase implements PostFilter {
 
   final ValueSourceRangeFilter rangeFilt;
-  boolean cache = true; // cache by default
   int cost;
```

Comment:
Do we still need this?

Reply:
No we don't need that anymore since we are extending ExtendedQueryBase. Thanks for highlighting this.

Reply:
Yes, I initially removed it and all the other stuff, but TestFiltering.testCache was failing. Upon investigating, I found out the test was failing because I removed them so I resulted to undo the changes

Reply:
That's weird.  Do you understand why the distinction of extending ExtendedQueryBase vis implementing ExtendedQuery caused that test to fail?  I could help look if you can't figure it out.

Reply:
Right now I'm really not sure why that is

Reply:
**Update:** Got some suggestions from @madrob that I think should fix it 
**re:** calling setCost in `FunctionRangeQuery` to access super class